.github/workflows/checks.yml

```diff
@@ -56,3 +56,36 @@ jobs:
       - name: Run tests in the project
         run: snforge test
         working-directory: myproject
+
+    test-action-with-tools-file:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v3
+
+        - name: Get scarb version
+          id: extractScarbVersion
+          run: |
+            snfoundry_version=$(curl -s https://api.github.com/repos/foundry-rs/starknet-foundry/releases/latest | grep tarball_url | awk -F '/' '{print $8}' | tr -d '",')
+            version=$(curl -s https://raw.githubusercontent.com/foundry-rs/starknet-foundry/$snfoundry_version/.tool-versions | awk '{print $2}')
+
+            echo "scarbVersion=$version" >> "$GITHUB_OUTPUT"
```

Comment:
I think this is unnecessary, we can just hard code some scarb version in .tool-versions and call it a day, as we do not need to check compatibilities between versions in this test.

To be more specific, my proposition is to:

- in one step create a .tool-versions file that will have bot scarb and starknet-foundry versions set to some specific values
- in another we setup scarb using the file
- in another we setup starknet-foundry using the file
- in last step we check the returned --version to make sure it is the expected one

sounds good?

Reply:
okay i simplified the test hopefully this is better 😅